packages/map-template/src/components/MapTemplate/MapTemplate.jsx

```diff
@@ -544,6 +547,21 @@ function MapTemplate({ apiKey, gmApiKey, mapboxAccessToken, venue, locationId, p
         setCategories(uniqueCategories);
     }
 
+    /*
+     * React on changes in the category prop.
+     * Check if the category property matches with any of the existing categories.
+     */
+    useEffect(() => {
+        if (mapsindoorsSDKAvailable) {
+            if (category) {
+                const existingCategory = categories.find((matched) => matched[0] === category)
```

Comment:
Can probably be shortened (not tested)

```suggestion
            if (category && categories.includes(matched => matched[0] === category) {
```